test/simple-asha

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import os
+import signal
+import sys
+
+import dbus
+import dbus.service
+import dbus.mainloop.glib
+
+import gi
+
+gi.require_version("Gst", "1.0")
+gi.require_version("GLib", "2.0")
+from gi.repository import GLib, Gst
+
+import bluezutils
+
+mainloop = None
+pipeline = None
+seqnum: int = 0
+
+
+def signal_handler(_sig, _frame):
+    print("Got interrupt")
+    mainloop.quit()
+
+
+signal.signal(signal.SIGINT, signal_handler)
+
+
+def usage():
+    print(f"Usage: simple-asha <remote addr> <audio file name> (optional volume 0-127)")
+
+
+def start_playback(fd: int, omtu: int):
+    global mainloop, pipeline
+    pktsize = 161
+
+    if omtu < pktsize:
+        print("Weird mtu", omtu)
+
+    outdata = bytearray(pktsize)
+
+    Gst.init(None)
+
+    pipeline = Gst.parse_launch(
+        f"""
+          filesrc location="{sys.argv[2]}" ! decodebin !
+          audioconvert ! audioresample !
+          audiobuffersplit output-buffer-duration="20/1000" ! avenc_g722 !
+          appsink name=sink emit-signals=true
+    """
+    )
+
+    def on_new_sample(sink):
+        global seqnum
+
+        sample = sink.emit("pull-sample")
+        buf = sample.get_buffer()
+
+        with buf.map(Gst.MapFlags.READ) as info:
```

Comment:
This doesn't work for me... My version of gst is returning a tuple from Gst.Memory.Map

https://gstreamer.freedesktop.org/documentation/gstreamer/gstmemory.html?gi-language=python#gst_memory_map

This is the code I got working... you may need some sort of hybrid  of both cases:
```python
    def on_new_sample(sink):
        global seqnum

        sample = sink.emit("pull-sample")
        buf = sample.get_buffer()

        _, info = buf.map(Gst.MapFlags.READ)

        pos = 0

        if info.size != pktsize - 1:
            print("Unexpected buffer size: ", info.size)

        outdata[pos] = seqnum % 256
        pos += 1

        for byte in info.data:
            outdata[pos] = byte
            pos += 1

        try:
            n = os.write(fd, outdata)
            if n != pktsize:
                print("Wrote less than expected: ", n)
        except:
            return Gst.FlowReturn.ERROR
        finally:
            buf.unmap(info)

        seqnum += 1

        return Gst.FlowReturn.OK
```

Reply:
Hmm, do you have `gst-python` installed? We have [an override](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/main/subprojects/gst-python/gi/overrides/Gst.py?ref_type=heads#L724) that should return just the `MapInfo`.